lib/services/local/okta.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package local
+
+import (
+	"context"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend"
+	"github.com/gravitational/teleport/lib/services"
+	"github.com/gravitational/teleport/lib/services/local/generic"
+)
+
+const (
+	oktaImportRulePrefix      = "okta_import_rule"
+	oktaImportRuleMaxPageSize = 200
+	oktaAssignmentPrefix      = "okta_assignment"
+	oktaAssignmentMaxPageSize = 200
+)
+
+// OktaService manages Okta resources in the Backend.
+type OktaService struct {
+	importRuleSvc *generic.Service[types.OktaImportRule]
+	assignmentSvc *generic.Service[types.OktaAssignment]
+}
+
+// NewOktaService creates a new OktaService.
+func NewOktaService(backend backend.Backend) (*OktaService, error) {
+	importRuleSvc, err := generic.NewService(&generic.ServiceConfig[types.OktaImportRule]{
+		Backend:       backend,
+		PageLimit:     oktaImportRuleMaxPageSize,
+		ResourceKind:  types.KindOktaImportRule,
+		BackendPrefix: oktaImportRulePrefix,
+		MarshalFunc:   services.MarshalOktaImportRule,
+		UnmarshalFunc: services.UnmarshalOktaImportRule,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	assignmentSvc, err := generic.NewService(&generic.ServiceConfig[types.OktaAssignment]{
+		Backend:       backend,
+		PageLimit:     oktaAssignmentMaxPageSize,
+		ResourceKind:  types.KindOktaAssignment,
+		BackendPrefix: oktaAssignmentPrefix,
+		MarshalFunc:   services.MarshalOktaAssignment,
+		UnmarshalFunc: services.UnmarshalOktaAssignment,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &OktaService{
+		importRuleSvc: importRuleSvc,
+		assignmentSvc: assignmentSvc,
+	}, nil
+}
+
+// ListOktaImportRules returns a paginated list of all Okta import rule resources.
+func (o *OktaService) ListOktaImportRules(ctx context.Context, pageSize int, nextToken string) ([]types.OktaImportRule, string, error) {
```

Comment:
Correct me if I'm wrong, but all these methods that does nothing more than just passing arguments to the underlying generic service are needed just to satisfy `OktaImportRules` interface?

Reply:
That's correct.